clients/unshaded/src/main/java/tachyon/client/file/AbstractTachyonFileSystem.java

```diff
@@ -66,19 +65,20 @@ protected AbstractTachyonFileSystem() {
   }
 
   @Override
-  public long create(TachyonURI path, CreateOptions options) throws IOException, TachyonException {
+  public long create(TachyonURI path, CreateOptions options)
+      throws FileAlreadyExistsException, IOException, InvalidPathException, TachyonException {
     FileSystemMasterClient masterClient = mContext.acquireMasterClient();
     try {
-      long fileId =
-          masterClient.createFile(path.getPath(), options.getBlockSize(), options.isRecursive(),
-              options.getTTL());
-      return fileId;
-    } catch (BlockInfoException e) {
-      throw new TachyonException(e, TachyonExceptionType.FILE_ALREADY_EXISTS);
-    } catch (FileAlreadyExistException e) {
-      throw new TachyonException(e, TachyonExceptionType.FILE_ALREADY_EXISTS);
-    } catch (InvalidPathException e) {
-      throw new TachyonException(e, TachyonExceptionType.INVALID_PATH);
+      return masterClient.createFile(path.getPath(), options.getBlockSize(), options.isRecursive(),
+          options.getTTL());
+    } catch (TachyonException e) {
+      if (e.getType() == TachyonExceptionType.BLOCK_INFO) {
+        throw new FileAlreadyExistsException(e.getMessage(), e);
```

Comment:
High-level question: should the unwrapping done by the `masterClient`? Since the whole tachyon exception type is used to isolate thrift exception from the rest of the codebase, shall we push the unwrapping down the boundary of the client?